### ISSUE TITLE
High failed calls

### ISSUE THREAD

**Issue by yifengyuan** [nanoporetech/modkit#116]
Hi, 

I am using guppy+remora+modkit to detect a deaza DNA modification on G. This modification was successfully detected by Tombo before. I have two samples of DNA, modification minus control and modification-containing DNA. 

I first used guppy for basecalling with mv tag. Then I used remora prepare, make_config, train and infer commands to generate control.bam and mod.bam. Finally, I used modkit to convert bam to bed file with threshold of 0.9.

I got mod.bam file as below. You can notice that the column 16 (N fail) is very high. I am wondering where could probably go wrong. Thank you.

> pBAD24dpdAVH7D	3	4	q	1	+	3	4	255,0,0	1 100.00 1 0 0 0 7938 64 0
> pBAD24dpdAVH7D	6	7	q	1	+	6	7	255,0,0	1 100.00 1 0 0 11 7975 19 0
> pBAD24dpdAVH7D	13	14	q	1	+	13	14	255,0,0	1 100.00 1 0 0 13 7969 30 0
> pBAD24dpdAVH7D	15	16	q	2	+	15	16	255,0,0	2 100.00 2 0 0 44 7902 67 0
> pBAD24dpdAVH7D	19	20	q	3	+	19	20	255,0,0	3 100.00 3 0 0 34 7913 68 0
> pBAD24dpdAVH7D	21	22	q	1	+	21	22	255,0,0	1 100.00 1 0 0 49 41 7929 0
> pBAD24dpdAVH7D	22	23	q	3	+	22	23	255,0,0	3 100.00 3 0 0 101 125 7792 0
> pBAD24dpdAVH7D	23	24	q	1	+	23	24	255,0,0	1 100.00 1 0 0 22 44 7955 0
> pBAD24dpdAVH7D	26	27	q	88	+	26	27	255,0,0	88 100.00 88 0 0 43 7819 75 0
> pBAD24dpdAVH7D	27	28	q	115	+	27	28	255,0,0	115 100.00 115 0 0 24 7843 43 0

The settings that I used.
guppy 6.5.7
hac model

remora dataset prepare
--refine-kmer-level-table 9mer_levels_v1.txt --refine-rough-rescale --refine-scale-iters 3

remora dataset make_config
default

remora model train
--model remora/models/ConvLSTM_w_ref.py --chunk-context 50 50 --seed 123

remora infer from_pod5_and_bam
--model model_best.pt

Thank you very much.

**Comment by marcus1487** [comment]
I'm going to move this issue over to remora as this seems to be more of a model training and application issue than downstream processing (primary modkit tasks). 

I have a couple of recommendations. First I would convert from Guppy to Dorado. We have not tested Remora models with Guppy for some time, so I would remove this variable from any potential issues. Next I would recommend leaving `--refine-scale-iters` with the default of -1. We have found that performing re-squiggling can be detrimental to the performance of Remora models without a number of parameter sweeps. I can go into these if this is required, but the move table usually provides the "right" bit of signal.

After these adjustments it would be good if you could describe your training data in a bit more detail. How exactly are you annotating the ground truth modified base positions in your training data? Do you expect false training examples to be present? 

Then I would want to see how training progressed? Did the model learn to call the training data well?

**Comment by yifengyuan** [comment]
Thank you very much for the prompt reply. I re-ran Remora pipeline with only one change: 
Previous:
dataset prepare --refine-kmer-level-table 9mer_levels_v1.txt --motif G 0 --mod-base q 7qG --refine-rough-rescale --refine-scale-iters 3
Current:
dataset prepare --refine-kmer-level-table 9mer_levels_v1.txt --motif G 0 --mod-base q 7qG --refine-rough-rescale --refine-scale-iters -1

I got a similar result in which the failed calls are dominant:
> pBAD24dpdAVH7D	3	4	q	1	+	3	4	255,0,0	1 100.00 1 0 0 0 7938 64 0
> pBAD24dpdAVH7D	13	14	q	2	+	13	14	255,0,0	2 100.00 2 0 0 12 7973 26 0
> pBAD24dpdAVH7D	15	16	q	1	+	15	16	255,0,0	1 100.00 1 0 0 47 7916 51 0
> pBAD24dpdAVH7D	19	20	q	13	+	19	20	255,0,0	13 100.00 13 0 0 31 7868 107 0
> pBAD24dpdAVH7D	21	22	q	2	+	21	22	255,0,0	2 100.00 2 0 0 118 104 7797 0
> pBAD24dpdAVH7D	22	23	q	14	+	22	23	255,0,0	14 100.00 14 0 0 339 558 7111 0
> pBAD24dpdAVH7D	23	24	q	15	+	23	24	255,0,0	15 100.00 15 0 0 152 150 7706 0
> pBAD24dpdAVH7D	24	25	q	7	+	24	25	255,0,0	7 100.00 7 0 0 148 9 7860 0

I really appreciate your advice. I can try to explain.
> 1) convert from Guppy to Dorado.
I am using a 15 CPU HPC. I thought that Dorado requires a GPU, so I used Guppy.

> 2) How exactly are you annotating the ground truth modified base positions in your training data?
I have not done the validation yet. We have the quantification of this modification using LC-MS. We know that this modification occurs on G without a consensus motif. When they occur on restriction enzyme cleavage sites such as GGATCC, the cutting is blocked. After comparison of two agarose gels, we can know if some restriction enzyme cleavage sites are modified or not. I am thinking that those sites can be used as ground truth.

> 3) Do you expect false training examples to be present?
I would say no.

> 4) Then I would want to see how training progressed? Did the model learn to call the training data well?
I am not sure how to tell. I have to do some research and teach myself before I can answer it.

Thank you very much for your time!
